analysers/merge_bicycle_parking_FR_strasbourg.py

```diff
@@ -0,0 +1,97 @@
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Merge import Analyser_Merge_Point, SourceOpenDataSoft, CSV, Load_XY, Conflate, Select, Mapping
+
+tag_mapping = {
+    "ref": (
+        lambda res["id_local"]
+    ),
+    "capacity": (
+        lambda res: None if res["capacite"] in (None, "0") else res["capacite"]
+    ),
+    "capacity:cargo_bike": (
+        lambda res: None if res["capacite_cargo"] in (None, "0") else res["capacite_cargo"]
+    ),
+#    "bicycle_parking": (
+#      type_accroche
+#      mobilier
+#      protection
+#    ),
+#    "access": (
+#        acces      
+#    ),
+    "fee": (
+        lambda res: "no" if res["gratuit"] == "vrai" else None
+    ),
+    "covered": (
+        lambda res: "yes" if res["couverture"] == "vrai" else None
+    ),
+    "surveillance": (
+        lambda res: "yes" if res["surveillance"] == "vrai" else None
+    ),
+    "lit": (
+        lambda res: "yes" if res["lumiere"] == "vrai" else None
+    ),
+    "website": (
+        lambda res["url_info"]
+    ),
+    "start_date": (
+        lambda res["d_service"]
+    ),
+    "owner": (
+        lambda res["proprietaire"]
+    ),
+    "operator": (
+        lambda res: "Eurométropole de Strasbourg" if res["gestionnaire"] == "eurometropole" else None
+    ),
+    "note": (
+        lambda res["commentaires"]
+    )
+}
+
+class Analyser_Merge_Bicycle_Parking_FR_Strasbourg(Analyser_Merge_Point):
+    def __init__(self, config, logger = None):
+        Analyser_Merge_Point.__init__(self, config, logger)
+        self.def_class_missing_official(item = 8150, id = 21, level = 3, tags = ['merge', 'public equipment', 'bicycle', 'fix:survey', 'fix:picture'],
+            title = T_('Strasbourg bicycle parking not integrated'))
+
+        self.init(
+            "https://data.strasbourg.eu/explore/dataset/stationnementcyclable_ville_et_eurometropole_de_strasbourg/information/",
+            "Stationnement cyclable au format BNSC",
+            CSV(SourceOpenDataSoft(
+                attribution="Ville et eurométropole de Strasbourg",
+                url="https://data.strasbourg.eu/explore/dataset/stationnementcyclable_ville_et_eurometropole_de_strasbourg/")),
+            Load_XY("geo_point_2d", "geo_point_2d",
+                xFunction = lambda x: Load_XY.float_comma(x.split(',')[1]),
+                yFunction = lambda y: Load_XY.float_comma(y.split(',')[0])
+            ),
+            Conflate(
+                select = Select(
+                    types = ["nodes", "ways"],
+                    tags = {"amenity": "bicycle_parking"}),
+                conflationDistance = 20,
+                mapping = Mapping(
+                    static1 = {"amenity": "bicycle_parking"},
+                    static2 = {"source": self.source},
+                    mapping1 = tag_mapping )))
```

Comment:
https://github.com/osm-fr/osmose-backend/blob/master/doc/4-Merge.md#mapping

I guess not all tags should go to `mapping1`. When there is change in mapping1 it popup again on Osmose-frontend. But ignored when tag change is on `mapping2`. The same for static1 and static2.

It is important to avoid flooding the contributors with boring issues.

Reply:
I get what you mean, I didn’t thought about it at all and simply copied what’s been done and added what seemed right. Which tags do you think would better fit into `mapping2`?

Reply:
Following further reading of the doc, I can’t see how any of the tags I put into `mapping1` could be some kind of "metadata". In my opinion they’re all necessary and any change made would mean some kind of important modification to be edited.

Reply:
Personally, I will limit mapping1 to ref, capacity* (and bicycle_parking). It is already what other merge contents.

Reply:
Fine by me!